test/integration/connect/envoy/main_test.go

```diff
@@ -0,0 +1,67 @@
+package envoy
+
+import (
+	"os"
+	"os/exec"
+	"testing"
+)
+
+func TestEnvoy(t *testing.T) {
+	var testcases = []string{
+		"case-badauthz",
+		"case-basic",
+		"case-centralconf",
+		"case-cfg-resolver-dc-failover-gateways-none",
+		"case-cfg-resolver-dc-failover-gateways-remote",
+		"case-cfg-resolver-defaultsubset",
+		"case-cfg-resolver-subset-onlypassing",
+		"case-cfg-resolver-subset-redirect",
+		"case-cfg-resolver-svc-failover",
+		"case-cfg-resolver-svc-redirect-http",
+		"case-cfg-resolver-svc-redirect-tcp",
+		"case-consul-exec",
+		"case-dogstatsd-udp",
+		"case-gateways-local",
+		"case-gateways-remote",
+		"case-gateway-without-services",
+		"case-grpc",
+		"case-http",
+		"case-http2",
+		"case-http-badauthz",
+		"case-ingress-gateway-http",
+		"case-ingress-gateway-multiple-services",
+		"case-ingress-gateway-simple",
+		"case-ingress-mesh-gateways-resolver",
+		"case-multidc-rsa-ca",
+		"case-prometheus",
+		"case-statsd-udp",
+		"case-stats-proxy",
+		"case-terminating-gateway-simple",
+		"case-terminating-gateway-subsets",
+		"case-terminating-gateway-without-services",
+		"case-upstream-config",
+		"case-wanfed-gw",
+		"case-zipkin",
```

Comment:
Not a big deal but any reason why this is hard coded rather than just using the directories present? Are you imagining it as a first step towards moving the test code into Golang?

Reply:
Great question, and that is exactly the reason. I considered making this list the directories, but I suspect that will make it harder to port things over. I imagine one of the next steps will be to port some of the test case setup/teardown logic into Go. At that point we may have multiple `TestX` functions which different lists, for tests that use more of the Go helpers, vs ones that use all bash.